src/terminal/adapter/adaptDispatchGraphics.cpp

```diff
@@ -259,6 +259,38 @@ bool AdaptDispatch::SetGraphicsRendition(const VTParameters options)
     return true;
 }
 
+// Routine Description:
+// - DECSCA - Modifies the character protection attribute. This operation was
+//   originally intended to support a range of logical character attributes,
+//   but the protected attribute was the only one ever implemented.
+// Arguments:
+// - options - An array of options that will be applied in order.
+// Return Value:
+// - True.
+bool AdaptDispatch::SetCharacterProtectionAttribute(const VTParameters options)
+{
+    auto& textBuffer = _api.GetTextBuffer();
+    auto attr = textBuffer.GetCurrentAttributes();
+    for (size_t i = 0; i < options.size(); i++)
+    {
+        const LogicalAttributeOptions opt = options.at(i);
+        switch (opt)
+        {
+        case Default:
+            attr.SetProtected(false);
+            break;
+        case Protected:
+            attr.SetProtected(true);
+            break;
+        case Unprotected:
+            attr.SetProtected(false);
+            break;
+        }

```

Comment:
Hmm. I like how clear this is, but it makes me wonder if it shouldn't be `attr.SetProtected(opt == Protected)` for some reason. I have no reason to request it specifically other than aesthetics, so f.f. to ignore :)

Reply:
Ah, of course. Thanks!